cmd/shellNode.go

```diff
@@ -0,0 +1,210 @@
+		hostname := *instance.PrivateDnsName
+
+		sendPublicKey(availabilityZone, instanceID, getSSHPublicKey(), sess)
+
+		jump, err := getJump()
+
+		if err != nil {
+			log.Fatal("Error encountered when finding jumpbox:", err.Error())
+		}
+
+		execSSHCommand(hostname, jump)
+
+		return nil
+	},
+}
+
+func getAWSRegionFromKubeconfig() (string, error) {
+	kubeconfig, err := getKubeconfig()
+	if err != nil {
+		return "", err
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return "", err
+	}
+
+	host := config.Host
+	hostParts := strings.Split(host, ".")
+	region := hostParts[len(hostParts)-4]
+
+	return region, nil
+}
+
+func execSSHCommand(hostname string, jump string) {
+	sshBinary, err := exec.LookPath("ssh")
+	if err != nil {
+		log.Fatal("Could not find ssh binary")
+	}
+
+	target := fmt.Sprintf("ec2-user@%s", hostname)
+	proxyJump := fmt.Sprintf("ProxyJump=%s", jump)
+	syscall.Exec(sshBinary, []string{"ssh", "-o", proxyJump, target}, os.Environ())
+}
+
+func getJump() (string, error) {
+	kubeconfigName, err := getKubeconfigName()
+	if err != nil {
+		return "", err
+	}
+
+	clusterEnv := strings.Replace(kubeconfigName, "kubeconfig-truss-", "", 1)
+	jumps := viper.GetStringMap("jumps")
+
+	return jumps[clusterEnv].(string), nil
+}
+
+func getSSHPublicKey() string {
+	homedir, err := os.UserHomeDir()
+
+	if err != nil {
+		log.Fatal("Error locating user homedir:", err.Error())
+	}
+
+	// TODO: Make this key configurable
+	publicKeyPath := fmt.Sprintf("%s/.ssh/id_rsa.pub", homedir)
+
+	publicKeyFile, err := ioutil.ReadFile(publicKeyPath)
+
+	if err != nil {
+		log.Fatalf("Unable to read public key from %s:", publicKeyFile)
+	}
+
+	return string(publicKeyFile)
+}
+
+func describeKubernetesNode(nodeName string) (*v1.Node, error) {
+	kubeconfig, err := getKubeconfig()
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	node, err := clientset.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return node, nil
+}
+
+func describeInstance(instanceID string, sess *session.Session) *ec2.Instance {
+	ec2svc := ec2.New(sess)
+
+	params := &ec2.DescribeInstancesInput{
+		InstanceIds: []*string{&instanceID},
+	}
+
+	resp, err := ec2svc.DescribeInstances(params)
+
+	if err != nil {
+		fmt.Println("there was an error listing instances in", err.Error())
+		log.Fatal(err.Error())
+	}
+
+	return resp.Reservations[0].Instances[0]
+}
+
+func sendPublicKey(availabilityZone string, instanceID string, publicKey string, sess *session.Session) {
+	svc := ec2instanceconnect.New(sess)
+
+	params := &ec2instanceconnect.SendSSHPublicKeyInput{
+		AvailabilityZone: aws.String(availabilityZone),
+		InstanceId:       aws.String(instanceID),
+		InstanceOSUser:   aws.String("ec2-user"),
```

Comment:
Again, might want to make the OS user configurable, here

Reply:
As long as we're using Amazon Linux, this shouldn't change, so I don't think it's worth making configurable at this point. I did dry this up a bit so it's only specified once.